frontend/packages/dev-console/integration-tests/features/topology/topology-chart-area-visual.feature

```diff
@@ -172,3 +172,86 @@ Feature: Topology chart area
               And user hovers on Add to Project and clicks on From Channel
               And user clicks on Create
              Then user is able to see different applications created from Samples, From Git, Container Image, From Dockerfile, From Devfile, From Catalog, Database, Operator Backed, Helm Charts, Event Source, Channel
+
+
+        @regression, @manual
+        Scenario: Upload JAR file form
+            Given user has a jar file named "sample_yaml_upload.yaml"
+              And user is at the Topology page
+             When user drags and drop jar file on topology
+             Then user sees Upload JAR file form
+              And user can see JAR section with jar file name with Browse and Clear button associated with it
+              And user can see Optional java commands, Runtime icon and Build image version under JAR section
+              And user can see General section with Application name and Name under it
+              And user can see Resources and Advanced options sections
+
+
+        @regression, @manual
+        Scenario: Drag and drop jar file in topology chart view
+            Given user has a jar file named "sample_yaml_upload.yaml"
+              And user is at the Topology page
+             When user drags and drop jar file on topology
+              And user gives Application name as "sample-upload-app" and workload Name as "sample-yaml-upload"
+              And user clicks on Create in Upload JAR file form
+             Then user is redirected to topology
+              And user can see a toast notification of JAR file uploading with link to build logs
+              And user can see deployment "sample-yaml-upload" in application "sample-upload-app" is created in topology
+
+
+        @regression @manual
+        Scenario: Add to Project to upload JAR file in topology
+            Given user is at the Topology page
+             When user right clicks on the empty chart area
+              And user clicks on Add to Project
+              And user clicks on Upload JAR file
+              And user clicks on Browse in JAR file section
+              And user selects file to upload
+              And user clicks Clear and reupload the file
+              And user gives Application name as "sample-upload-app" and workload Name as "sample-yaml-upload-1"
+              And user clicks on Create
+             Then user is redirected to topology
+              And user can see a toast notification of JAR file uploading with link to build logs
+              And user can see deployment "sample-yaml-upload-1" in application "sample-upload-app" is created in topology
+
+
+        @regression @manual
+        Scenario: Add to Project through drag and drop to upload JAR file in topology
```

Comment:
This scenario is a bit redundant with the one above it considering the difference has already been tested as part of the form tests. The interaction with topology is the same.

Reply:
I understand but we need to consider all the scenario as part of epic

Reply:
I suppose. I'm not sure of your overall strategy.
IMO testing topology doesn't require re-testing the different form flows. As long as one of the flows works.